src/server/pkg/pool/pool.go

```diff
@@ -109,7 +109,7 @@ func (p *Pool) Do(ctx context.Context, f func(cc *grpc.ClientConn) error) error
 				if mapConn.cc == nil {
 					cc, err := grpc.DialContext(ctx, addr, p.opts...)
 					if err != nil {
-						return err
+						return fmt.Errorf("failed to connect to %s: %+v", addr, err)
```

Comment:
Not sure if `%+v` is necessary for `err`.

Reply:
Good point, are you looking for `%v` or `%s` here though?

Reply:
`%v` is more idiomatic I think.